src/main/scala/it/pagopa/interop/notifier/service/impl/CatalogManagementServiceImpl.scala

```diff
@@ -11,7 +11,7 @@ import it.pagopa.interop.commons.logging.{CanLogContextFields, ContextFieldsToLo
 import java.util.UUID
 import scala.concurrent.{ExecutionContext, Future}
 
-final case class CatalogManagementServiceImpl(invoker: CatalogManagementInvoker, api: CatalogManagementApi)(implicit
```

Comment:
In our code base, we use `case classes` and `classes` for the same purpose (e.g. look at this project services before my change).
I suggest a common approach to their usage.
What I'm here suggesting, is to use simple classes to implement services, while I would use case classes for models.
I don't have a strong opinion about that, but I would choose a shared approach here and in all the other services.

Reply:
I agree to use a shared approach.
IMO case classes offer that kind of code sugar and utils worth the low (very low in case of services, instantiated just once) memory impact

Reply:
I'm with @beetlecrunch. I think that @galales might be referring to the overloaded apply method for construction, but that's a thing that can be easily replaced by writing a custom apply constructor if needed. `new` despite being ugly is a short and valid alternative.
Other case class features that might come in handy in the case of models are not used by services as far as I'm aware. I'm thinking about equality, serializability, ease of access to the fields of the constructor, unapply methods for destructuring, and hashcode. None of these is actually used by services so I don't see the point in having them as case classes.
Is there any other feature that we're in fact leveraging that I'm missing @galales?

Reply:
To conclude, the possible/plausible damage IMHO, is that we might leverage those methods and syntax sugar in situations that might introduce silent or difficult to debug behaviours that might cause a bug

Reply:
fine by me